### PR TITLE
Change App ID for PR builds of Android app

### PR DESCRIPTION
**Pull request by jakubgs**
We don't want to make the PR builds replace the release when installed, or have access to the accounts.

For that reason this PR changes the ID of Android app from `im.status.ethereum` to `im.status.ethereum.pr` and name from `Status` to `Status PR` for PR builds.

Resolves: https://github.com/status-im/status-react/issues/7286
Status: READY

### CODE REVIEW

**Comment by jakubgs** [comment]
It doesn't seem like `gradle` likes my changes:
```
Execution failed for task ':app:processReleaseGoogleServices'.
> No matching client found for package name 'im.status.ethereum.pr'
```

**Comment by mandrigin** [comment]
@jakubgs you maybe can use something like what we use for running dev versions: `react-native run-android --appIdSuffix debug`

**Comment by jakubgs** [comment]
Thanks @mandrigin I'll look into that. But I've been thinking that maybe the proper way of doing this is adding another `buildType` for android:
https://github.com/status-im/status-react/blob/develop/android/app/build.gradle#L195
There's a `release` and `debug`, maybe there should be also a `pr`. But I'm not yet sure how that is selected.
https://developer.android.com/studio/build/build-variants

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/status-im/status-react">status-im/status-react</a></strong></div><div>a free (libre) open source, mobile OS for Ethereum - status-im/status-react</div></blockquote>
<blockquote><div>Android Developers</div><div><strong><a href="https://developer.android.com/studio/build/build-variants">Configure build variants  |  Android Developers</a></strong></div></blockquote>

**Comment by jakubgs** [comment]
__OR__ I could just do the `.debug` suffix and call it a day. Not sure if we care about the name referencing PRs. We just want the ID to be different from release ones, right?

**Comment by mandrigin** [comment]
@jakubgs I guess we might need to update package name in `android/app/google-services.json` too...

**Comment by jakubgs** [comment]
>@jakubgs I guess we might need to update package name in android/app/google-services.json too...

I don't think that's correct. I think the ones we need to update are in `android/app/src/`, and add another type of build which is PR, and then run `gradlew assemblePr` instead of `gradlew assembleRelease`.

File: android/app/src/pr/res/values/strings.xml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<resources>
+    <string name="app_name">Status Debug</string>
```
Comment (mandrigin) [inline android/app/src/pr/res/values/strings.xml]: Status PR Build?

Comment (jakubgs) [reply android/app/src/pr/res/values/strings.xml]: Right.

File: ci/ios.groovy [hunk 2]
```diff
@@ -19,6 +19,10 @@ def bundle(type) {
   plutil('CFBundleShortVersionString', cmn.version())
   plutil('CFBundleVersion', cmn.genBuildNumber())
   plutil('CFBundleBuildUrl', currentBuild.absoluteUrl)
+  if (type == 'pr') { /* PR builds shouldn't replace normal releases */
+    env.BUNDLE_ID_SUFFIX = '.pr'
+    //plutil('CFBundleIdentifier', 'im.status.ethereum.pr')
```
Comment (mandrigin) [inline ci/ios.groovy]: commented-out line... is that expected?

Comment (jakubgs) [reply ci/ios.groovy]: Considering PR has `Status: WIP` in description, yes :D.

**Review by flexsurfer — APPROVED** [review]

**Review by pedropombeiro — APPROVED** [review]
lgtm

**Comment by jakubgs** [comment]
I've installed the [Android Build](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190209-131905-7ebdbd-pr.apk) and it showed up as a new application and did not have access to old accounts.
Can someone confirm for iOS?

Also, Both Status apps show up as "Status". Maybe the name should be also different. I'll look into it.

**Comment by mandrigin** [comment]
@jakubgs iOS has the same bundle ID (I tried build `#27`).

**Comment by jakubgs** [comment]
I've decided that doing the same change for iOS is going to be very messy so I'm limiting this PR just to Android changes so I can merge it. I'll do iOS separately.

**Comment by pedropombeiro** [comment]
lgtm

**Pull request merged**